source/event_stream_rpc.c

```diff
@@ -0,0 +1,113 @@
+
+        /* check type first since that's cheaper than a string compare */
+        if (header->header_value_type == AWS_EVENT_STREAM_HEADER_INT32) {
+
+            struct aws_byte_buf stream_id_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_stream_id_name.ptr, aws_event_stream_rpc_stream_id_name.len);
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &stream_id_field)) {
+                *stream_id = aws_event_stream_header_value_as_int32(header);
+                AWS_LOGF_DEBUG(AWS_LS_EVENT_STREAM_GENERAL, "stream id header value %" PRId32, *stream_id);
+                required_fields_found += 1;
+                goto found;
+            }
+
+            struct aws_byte_buf message_type_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_message_type_name.ptr, aws_event_stream_rpc_message_type_name.len);
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &message_type_field)) {
+                *message_type = aws_event_stream_header_value_as_int32(header);
+                AWS_LOGF_DEBUG(AWS_LS_EVENT_STREAM_GENERAL, "message type header value %" PRId32, *message_type);
+                required_fields_found += 1;
+                goto found;
+            }
+
+            struct aws_byte_buf message_flags_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_message_flags_name.ptr, aws_event_stream_rpc_message_flags_name.len);
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &message_flags_field)) {
+                *message_flags = aws_event_stream_header_value_as_int32(header);
+                AWS_LOGF_DEBUG(AWS_LS_EVENT_STREAM_GENERAL, "message flags header value %" PRId32, *message_flags);
+                required_fields_found += 1;
+                goto found;
+            }
+        }
+
+        if (header->header_value_type == AWS_EVENT_STREAM_HEADER_STRING) {
+            struct aws_byte_buf operation_field = aws_byte_buf_from_array(
+                aws_event_stream_rpc_operation_name.ptr, aws_event_stream_rpc_operation_name.len);
+
+            if (aws_byte_buf_eq_ignore_case(&name_buf, &operation_field)) {
+                *operation_name = aws_event_stream_header_value_as_string(header);
+                AWS_LOGF_DEBUG(
+                    AWS_LS_EVENT_STREAM_GENERAL,
+                    "operation name header value" PRInSTR,
+                    AWS_BYTE_BUF_PRI(*operation_name));
+                optional_fields_found += 1;
+                goto found;
+            }
+        }
+
+        continue;
+
+    found:
+        if (required_fields_found == 3 && optional_fields_found == 1) {
+            return AWS_OP_SUCCESS;
+        }
+    }
+
+    AWS_LOGF_TRACE(
+        AWS_LS_EVENT_STREAM_GENERAL,
+        "required fields found %zu, optional fields found %zu.",
+        required_fields_found,
+        optional_fields_found);
+    return required_fields_found == 3 ? AWS_OP_SUCCESS : AWS_OP_ERR;
```

Comment:
need to aws_raise_error() in case of failure.

Reply:
also, this is not a good way to check that we got all required fields. If There could just be 3 copies of the ":message-type" header, and we'd pass even though there were no ":message-flags" or ":stream-id"